src/components/IdScanner/IdScanner.tsx

```diff
@@ -1,63 +1,121 @@
 import React, { FunctionComponent, useState, useEffect } from "react";
-import { StyleSheet, View } from "react-native";
+import { Dimensions, LayoutRectangle, StyleSheet, View } from "react-native";
 import * as Permissions from "expo-permissions";
 import { color, size } from "../../common/styles";
-import { BarCodeScanner, BarCodeScannerProps } from "expo-barcode-scanner";
-import { SecondaryButton } from "../Layout/Buttons/SecondaryButton";
+import {
+  BarCodeScannedCallback,
+  BarCodeScanner,
+  BarCodeScannerProps
+} from "expo-barcode-scanner";
 import { LoadingView } from "../Loading";
+import { LightBox } from "../Layout/LightBox";
+import { Ionicons } from "@expo/vector-icons";
+import { TransparentButton } from "../Layout/Buttons/TransparentButton";
+import { IdScannerLabel } from "./IdScannerLabel";
 
 const styles = StyleSheet.create({
   cameraWrapper: {
-    position: "absolute",
-    top: 0,
-    left: 0,
-    right: 0,
-    bottom: 0,
-    alignItems: "center",
-    justifyContent: "center",
-    backgroundColor: color("grey", 0)
+    ...StyleSheet.absoluteFillObject,
+    backgroundColor: color("grey", 100)
   },
-  cancelButtonWrapper: {
-    marginTop: size(3),
-    marginBottom: size(4)
+  backButtonWrapper: {
+    marginTop: size(3)
+  },
+  scanner: {
+    ...StyleSheet.absoluteFillObject
+  },
+  interestArea: {
+    position: "absolute"
   }
 });
 
+const interestAreaRatios: Record<string, Record<string, number>> = {
+  [BarCodeScanner.Constants.BarCodeType.qr]: { width: 0.7, height: 0.35 },
+  [BarCodeScanner.Constants.BarCodeType.code39]: { width: 0.9, height: 0.2 }
+};
+
+const getInterestAreaDimensions = (
+  barCodeTypes: string[] | undefined
+): LayoutRectangle => {
+  const { width, height } = Dimensions.get("window");
+  if (!barCodeTypes) {
+    return { x: 0, y: 0, width, height };
+  }
+  let maxWidthRatio = 0;
+  let maxHeightRatio = 0;
+  barCodeTypes.forEach(type => {
+    maxWidthRatio = Math.max(maxWidthRatio, interestAreaRatios[type].width);
+    maxHeightRatio = Math.max(maxHeightRatio, interestAreaRatios[type].height);
+  });
+  return {
+    x: (width * (1 - maxWidthRatio)) / 2,
+    y: (height * (1 - maxHeightRatio)) / 2,
+    width: width * maxWidthRatio,
+    height: height * maxHeightRatio
+  };
+};
+
 export type Camera = Pick<
   BarCodeScannerProps,
   "onBarCodeScanned" | "barCodeTypes"
->;
+> & {
+  cancelButtonText?: string;
+  onCancel?: () => void;
+  hasLimitedInterestArea?: boolean;
```

Comment:
I think hasLimitedInterestArea is not necessary, you can just check if interestAreaLayout is there!

Reply:
I have removed `hasLimitedInterestArea` here!